summernote/summernote
# Fixe for issue #3948

#### What does this PR do?

- fixes issue #3948 

#### Where should the reviewer start?

- start on the config/scss.config.js

#### How should this be manually tested?

After running `npm run build`, the **dist** folder contains several _.css_ files. Open one of the files in a text editor and search for the string `before{content:` if there is a square rectangle in the content then the generated character is an UTF-8 code and this will confuse browsers when selecting the font and the glyph to be rendered from the font.

After applying the fix, a new process has been added to the **postcss** processor to re-escape the UTF-8 characters so you will see content:"\ea01" instead of the square rectangle. In other words, the content will be exactly the same as it is in the original _summernote.scss_ file.

#### Any background context you want to provide?

The source of the issue is the sass processor that is part of the sass-loader that is converting the escaped characters into the non escaped equivalents in UTF-8. Sadly, there are long discussions about this topic in mozz and other places.

The spec 
[https://drafts.csswg.org/css-syntax/#determine-the-fallback-encoding](https://drafts.csswg.org/css-syntax/#determine-the-fallback-encoding)
is not straight forward about what charset a css has to be interpreted and as a consequence, different browsers implement different implementations.

The proposed solution is the safest possible solution that means the content of the css file will always by pure ASCII compatible with UTF-8 and with ISO-8859-1. The content is always escaped to avoid implementation issues.

### Checklist

In order for the new post processor to success it is required to update postcss-loader to version 4.1.0 as the version 3.0.0 cannot read the configuration from inside the _scss.config.js_ file
